src/trace/trace.cpp

```diff
@@ -479,6 +479,21 @@ otf2::writer::local& Trace::bio_writer(BlockDevice dev)
     return archive_(intern_location);
 }
 
+otf2::writer::local& Trace::topdown_writer(const ExecutionScope& writer_scope)
+{
+    MeasurementScope scope = MeasurementScope::topdown(writer_scope);
+
+    const auto& intern_location = registry_.emplace<otf2::definition::location>(
+        ByMeasurementScope(scope), intern(scope.name()),
+        registry_.get<otf2::definition::location_group>(
+            ByExecutionScope(groups_.get_parent(writer_scope))),
+        otf2::definition::location::location_type::cpu_thread);
+
+    comm_locations_group_.add_member(intern_location);
```

Comment:
Why adding the location to `comm_locations_group_`? Without that this function is identical to 
`metric_writer(MeasurementScope::topdown(scope))`.